packages/server/src/index.ts

```diff
@@ -22,38 +19,14 @@ app.get('/hello', (req, res) => {
   res.send({ express: 'Hello From Express' });
 });
 
+// An auth endpoint for GitHub that returns either `{ access_token: string }` or `{ error: string }`
 app.post('/auth', (req, res) => {
   const { code, state } = req.body;
-  request.post(
-    {
-      url: 'https://github.com/login/oauth/access_token',
-      headers: {
-        Accept: 'application/json',
-      },
-      json: {
-        client_id: GITHUB_CLIENT_ID,
-        client_secret: GITHUB_CLIENT_SECRET,
-        redirect_uri: GITHUB_REDIRECT_URL,
-        code,
-        state,
-      },
-    },
-    (error, httpResponse, body) => {
-      // TODO: pretty sure a 404 went into the else of this if
-      if (error) {
-        console.error('github login failed');
-        res
-          .contentType('application/json')
-          .status(500)
-          .send({ error });
-      } else {
-        res
-          .contentType('application/json')
-          .status(200)
-          .send(body);
-      }
-    },
-  );
+  respondWithAccessToken({
```

Comment:
I do not like how res gets passed to respondWithAccessToken. I think that it should be this function's job to get anything out of req that is needed, and send back whatever is needed with res.

Reply:
Yeah, fair point.  Traded it out to a Promise, and reduced its scope to only produce the payload, not send it.